### PR TITLE
add customer comments + inventory reports

### PR DESCRIPTION
**Pull request by RyanfaeScotland**
Some extra useful reports and a bug fix for #247

### CODE REVIEW

File: application/language/en/reports_lang.php [hunk 1]
```diff
@@ -6,6 +6,8 @@
 $lang["reports_categories_summary_report"] = "Categories Summary Report";
 $lang["reports_category"] = "Category";
 $lang["reports_comments"] = "Comments";
+$lang["reports_comments_customer"] = "Customer Comments";
+$lang["reports_comments_customer_report"] = "Customer Comments Report";
```
Comment (jekkos) [inline application/language/en/reports_lang.php]: Did you change the language csv file first? These files are auto generated with the generate_langauges.php script in the root of the project.

Comment (RyanfaeScotland) [inline application/language/en/reports_lang.php]: No I edited them direct. Feel free to reject and I'll redo. Do we have instructions written up for this process or will I write them as I go?

Comment (jekkos) [inline application/language/en/reports_lang.php]: Might be good indeed to have some reference about this, as new people often edit the generated files instead of the CSV's. Just a matter of opening them in excel and then editing the columns for each language.

afterwards the script has to be run as foillows

     php generate_languages.php

Comment (RyanfaeScotland) [inline application/language/en/reports_lang.php]: Never hurts to have auto-generated files declare themselves as such with a little auto-generated line at the top of the file. Will look to get on it this weekend.

File: application/models/reports/inventory_facebook.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		return array($this->lang->line('reports_item_name'),
+					$this->lang->line('reports_category'),
+					$this->lang->line('reports_count'),					
+					$this->lang->line('reports_unit_price'));
+	}
+	
+	public function getData(array $inputs)
+	{
+	    $this->db->from('items');
+        $this->db->join('item_quantities','items.item_id=item_quantities.item_id');
```
Comment (daN4cat) [inline application/models/reports/inventory_facebook.php]: Would be good to have the same indentation

Comment (RyanfaeScotland) [inline application/models/reports/inventory_facebook.php]: I think there is a file I need to add that sets my indentation to be updated automatically. Now that I know my commits can be merged with others I'll focus more on keeping myself consistent with you guys.

File: application/models/item.php [hunk 3]
```diff
@@ -49,7 +49,7 @@ public function search($search, $filters, $rows=0, $limit_from=0)
 	{
 		$this->db->from('items');
 		$this->db->join('suppliers', 'suppliers.person_id = items.supplier_id', 'left');
-		$this->db->join('inventory', 'inventory.trans_items = items.item_id');
+		$this->db->join('inventory', 'inventory.trans_items = items.item_id', 'left');
```
Comment (jekkos) [inline application/models/item.php]: Might be better to make a separate pr for this issue. You can do that by adding this change on a separate branch, starting from master.

File: application/views/reports/listing.php [hunk 4]
```diff
@@ -51,6 +52,7 @@
 		<?php 
 			show_report('', 'reports_inventory_low');	
 			show_report('', 'reports_inventory_summary');
+			show_report('', 'reports_inventory_facebook');
```
Comment (jekkos) [inline application/views/reports/listing.php]: maybe rename the last report to inventory_customer? Which extra fields does it show in the report? If I find the time later today then I'll have a closer look.

File: application/models/item.php (hunk 3 above)
Comment (jekkos) [inline application/models/item.php]: Otherwise if too hard I can pull in your master and cherry-pick this change + the typo correction already. But then you'll have the risk of running into trouble when syncing back with my repo afterwards.

File: application/models/reports/inventory_facebook.php (hunk 2 above)
Comment (jekkos) [inline application/models/reports/inventory_facebook.php]: Think ospos is using spaces as indentation currently, similary as in codeigniter. Might be good to have git convert these automatically.

File: application/views/reports/listing.php (hunk 4 above)
Comment (RyanfaeScotland) [inline application/views/reports/listing.php]: Yeah I was thinking that as well but needed it to be clear for my none-techie partner who was also using the system and at that point I wasn't sure how the naming worked. Going to get that revised.

Annoyingly (very annoyingly) I've already had to shut the business down due to accounting issues. I'll still commit to this because I think it is a great project but I'll have to deal with the fall out from the business first.

**Comment by daN4cat** [comment]
What do we want to do with this PR?

Not sure we want to include this report or not, I'm even not sure what the purpose for it was...

**Comment by RyanfaeScotland** [comment]
Hey @daN4cat , completely up to you guys of course but here was my thoughts behind it. We we're using the comments to record customer requests for specific items and once a week we'd print off the report and see if we could source any of them.

Eventually I planned to have a requested items field that would let me mark items from our potential inventory (i.e. items that are out of stock but may come back in) to a customer so that it would automatically flag when an item a customer was looking for came back in but in the meantime this manual solution was doing the job.

**Comment by daN4cat** [comment]
I suggest to close this PR and not take the code in. The use case would be interesting if fully automated, this solution is manual and seems to me a bit ad hoc.
